controllers/activemqartemis_controller_test.go

```diff
@@ -3871,252 +3876,341 @@ var _ = Describe("artemis controller", func() {
-			loggingSecret := secrets.NewSecret(types.NamespacedName{Name: loggingSecretName, Namespace: defaultNamespace}, loggingSecretName, loggingData, nil)
-			Eventually(func() bool {
-				err := k8sClient.Create(ctx, loggingSecret, &client.CreateOptions{})
-				return err == nil
-			}, timeout, interval).Should(BeTrue())
+			_, isConfigMap := resource.(*corev1.ConfigMap)
+			if isConfigMap {
+				crd.Spec.DeploymentPlan.ExtraMounts.ConfigMaps = []string{resource.GetName()}
+			} else {
+				crd.Spec.DeploymentPlan.ExtraMounts.Secrets = []string{resource.GetName()}
+			}
+			if test.shouldCreate {
+				Eventually(func() bool {
+					err := k8sClient.Create(ctx, resource, &client.CreateOptions{})
+					return err == nil
+				}, timeout, interval).Should(BeTrue())
+			}
 
-			By("By creating a new crd")
-			crd := generateArtemisSpec(defaultNamespace)
-			crd.Spec.DeploymentPlan.ExtraMounts.Secrets = []string{loggingSecretName}
+			// when
 			Expect(k8sClient.Create(ctx, &crd)).Should(Succeed())
 
+			// then
 			createdCrd := &brokerv1beta1.ActiveMQArtemis{}
-			Eventually(func() bool { return getPersistedVersionedCrd(crd.ObjectMeta.Name, defaultNamespace, createdCrd) }, timeout, interval).Should(BeTrue())
-			Expect(createdCrd.Name).Should(Equal(crd.ObjectMeta.Name))
-
 			Eventually(func(g Gomega) {
 				g.Expect(getPersistedVersionedCrd(crd.Name, defaultNamespace, createdCrd)).To(BeTrue())
 				validCondition := meta.FindStatusCondition(createdCrd.Status.Conditions, common.ValidConditionType)
 				g.Expect(validCondition).NotTo(BeNil())
-				g.Expect(validCondition.Status).To(Equal(metav1.ConditionFalse))
-				g.Expect(validCondition.Reason).To(Equal(common.ValidConditionFailedReason))
-				g.Expect(validCondition.Message).To(Equal(fmt.Sprintf("Logging secret %v must have key logging.properties", loggingSecret.Name)))
-			}, timeout, interval).Should(Succeed())
 
-			By("deleting the logging configmap")
-			Expect(k8sClient.Delete(ctx, loggingSecret)).Should(Succeed())
+				if errorMsg != "" {
+					g.Expect(validCondition.Status).To(Equal(metav1.ConditionFalse))
+					g.Expect(validCondition.Reason).To(Equal(common.ValidConditionMissingResourcesReason))
+					g.Expect(validCondition.Message).To(Equal(fmt.Sprintf("Missing required resources [%v]", errorMsg)))
+				} else {
+					g.Expect(validCondition.Status).To(Equal(metav1.ConditionTrue))
+					g.Expect(validCondition.Reason).To(Equal(common.ValidConditionSuccessReason))
+					g.Expect(validCondition.Message).To(BeEmpty())
+					validateSts(test, createdCrd)
+				}
+			}, timeout, interval).Should(Succeed())
 
+			if test.shouldCreate {
+				Expect(k8sClient.Delete(ctx, resource)).Should(Succeed())
+				Eventually(func(g Gomega) {
+					key := types.NamespacedName{
+						Name:      resource.GetName(),
+						Namespace: resource.GetNamespace(),
+					}
+					g.Expect(k8sClient.Get(ctx, key, test.buildResource())).To(BeFalse())
+				})
```

Comment:
Here Eventually(func(g Gomega){...}), should add timeout, interval).Should(Succeed())